model/VocabularyConfig.php

```diff
@@ -31,41 +31,97 @@ class VocabularyConfig extends BaseConfig
     public function __construct($resource, $globalPlugins=array())
     {
         $this->resource = $resource;
-        $plugins = $this->resource->allLiterals('skosmos:usePlugin');
+        $this->globalPlugins = $globalPlugins;
+        $this->setParametrizedPlugins();
+        $pluginArray = $this->getPluginArray();
+        $this->pluginRegister = new PluginRegister($pluginArray);
+    }
+
+    /**
+     * Get an ordered array of plugin names with order configured in skosmos:vocabularyPlugins
+     * @return array|null of plugin names
+     */
+    public function getPluginArray() {
+
         $pluginArray = array();
+
+        $vocabularyPlugins = $this->resource->getResource('skosmos:vocabularyPlugins');
+        if ($vocabularyPlugins) {
+            // $vocabularyPlugins has all resources
+            foreach ($vocabularyPlugins as $plugin) {
+                if ($plugin instanceof EasyRdf\Literal) {
+                    $pluginName = $plugin->getValue();
+                    array_push($pluginArray, $pluginName);
+                }
+                else {
+                    array_push($pluginArray, $plugin->getLiteral('skosmos:usePlugin')->getValue());
+                }
+            }
+        }
+        $pluginArray = array_merge($pluginArray, $this->globalPlugins);
+
```

Comment:
Rest could be sorted, now they are not. However, this is not required, just a note.

Implementation in short:

Use new array variable
Collect them there
Sort
Then merge and unique